### PR TITLE
Add verify pack script

### PR DESCRIPTION
**Pull request by steve1515**
Adding a verify_pack script that can be run in order to verify that a directory matches up correctly with a SMDB. This script can be run when wanting to verify all files in an already created pack directory without having to move any files. No files in the pack are moved/modified when this script is executed.

The script outputs the following information:
- Files in incorrect locations (i.e. files on disk with a hash in the SMDB, but in the wrong location)
- Extra files (i.e. files on disk with a hash not in the SMDB)
- Missing files (i.e. files in the SMDB, but not on disk)

### CODE REVIEW

**Comment by frederic-mahe** [comment]
hi @steve1515 thanks for this new script. I think what you suggest is a good idea, and if @SmokeMonsterPacks  and @tjanas agree, I will merge your pull request (after some code review).

File: verify_pack.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    number_missing_files = len(MISSING_FILES)
+    
+    # write information to log file only if there are any bad, extra or missing files to report
+    if MISMATCH_FILES and (number_bad_loc_files > 0 or number_extra_files > 0 or number_missing_files > 0):
+        BAD_LOCATION_FILES.sort()
+        EXTRA_FILES.sort()
+        MISSING_FILES.sort()
+        
+        with open(MISMATCH_FILES, "w") as mismatch_files:
+            if number_bad_loc_files > 0:
```
Comment (frederic-mahe) [inline verify_pack.py:196]: why not?
```py
            if BAD_LOCATION_FILES:
                print("Incorrect Location Files:", file=mismatch_files)
                for file, hash_sha256 in BAD_LOCATION_FILES:
                    print(os.path.abspath(file), hash_sha256, sep="\t", file=mismatch_files)
                print("\n", file=mismatch_files)
```

**Comment by tjanas** [comment]
I like this idea, so after code reviews and proposed changes, let’s add this new script.

**Comment by steve1515** [comment]
Updates have been made. Thanks for the suggestions!
I've also updated the ReadMe file so there is some documentation on the new script.

**Comment by frederic-mahe** [comment]
Thanks @steve1515 ! Now, I count on your support for the long-term maintenance of this new script :-)

**Pull request merged**